crates/utils/src/utils/validation.rs

```diff
@@ -299,6 +299,40 @@ pub fn check_url_scheme(url: &Option<Url>) -> LemmyResult<()> {
   }
 }
 
+pub fn block_url_regex(blocked_url: &str) -> LemmyResult<Regex> {
+  let blocked = Url::parse(blocked_url)?;
+  let block_regex = if blocked_url.ends_with('/') {
+    Regex::new(&format!(
+      "({}://)?{}{}?",
+      blocked.scheme(),
+      escape(blocked.domain().expect("No domain.")),
+      escape(blocked.path())
+    ))?
+  } else {
+    Regex::new(&format!(
+      "({}://)?{}{}",
+      blocked.scheme(),
+      escape(blocked.domain().expect("No domain.")),
+      escape(blocked.path())
+    ))?
+  };
+
+  Ok(block_regex)
+}
+
+pub fn is_url_blocked(url: &Option<Url>, blocklist: Vec<String>) -> LemmyResult<()> {
+  if let Some(url) = url {
+    for blocked_url in &blocklist {
+      let block_regex = block_url_regex(blocked_url)?;
```

Comment:
Compiling regex is quite inefficient so you should cache the whole `Vec<Regex>` using once_cell like [here](https://github.com/LemmyNet/lemmy/blob/main/crates/api_crud/src/site/read.rs#L33). 

Using a [RegexSet](https://docs.rs/regex/latest/regex/struct.RegexSet.html) is probably even more efficient (and then again caching it after compilation).